filebeat/input/filestream/input.go

```diff
@@ -158,6 +171,26 @@ func (inp *filestream) Run(
 	})
 	defer streamCancel()
 
+	inp.metrics.onReaderStarted()
+	defer inp.metrics.onReaderStopped()
+
+	if inp.detailedMetricsEnabled {
+		inp.detailedMetrics = newProgressMetrics(inp.ID, fs.newPath, src.Name())
+		go func() {
+			err := timed.Periodic(streamCtx, 30*time.Second, func() error {
+				fi, err := os.Stat(fs.newPath)
```

Comment:
`os.Stat` is always subject to race conditions. If possible we would need the update to be based on the file object.